src/update-client.cc

```diff
@@ -376,6 +379,119 @@ void update_client::do_stuff()
 	resolver.async_resolve( params->host.authority, params->host.scheme, cb);
 }
 
+void update_client::install_package(const std::string& packageName, std::string url, const std::string& startParams)
+{
+	installer_events->installer_download_start(packageName);	
+
+	boost::system::error_code error;
+	boost::asio::io_service io_service;
+
+	// Deduce domain from url
+	std::string domainName;
+	boost::replace_all(url, "https://", "");
+	boost::replace_all(url, "http://", "");
+
+	for (size_t itr = 0; itr < url.size(); ++itr)
+	{
+		if (url[itr] == '/' || url[itr] == '\\')
+			break;
+
+		domainName.push_back(url[itr]);
+	}
+
+	// Resolve domain to IP
+	tcp::resolver local_resolver(io_service);
+	tcp::resolver::iterator endpoint_iterator = local_resolver.resolve(tcp::resolver::query{ domainName, "https" }, error);
+
+	if (error.failed())
+	{
+		installer_events->installer_package_failed(packageName, "HTTP " + error.message());
+		return;
+	}
+
+	// Try connect each endpoint until success
+	ssl::stream<tcp::socket> local_ssl_socket(io_ctx, ssl_context);	
+	
+	do
+	{
+		local_ssl_socket.lowest_layer().close();
+		local_ssl_socket.lowest_layer().connect(*endpoint_iterator++, error);
+	}	
+	while (error && endpoint_iterator != tcp::resolver::iterator{});
```

Comment:
Should we have a timeout here to make sure this never ends up in an infinite loop or a very long wait?